### PR TITLE
minor refactor in istio role

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
Minor changes.

### CODE REVIEW

File: roles/istio/defaults/main.yaml [hunk 1]
```diff
@@ -5,6 +5,5 @@ istio_create_seldon_gateway: true
 istio_ssl:
   enabled: true
   self_signed_ca: true
-#  server_crt_path: /path/to/server.crt
-#  server_key_path: /path/to/server.key
-
+  server_crt_path: null
```
Comment (RafalSkolasinski) [inline roles/istio/defaults/main.yaml:8]: This way more clear that it can be provided

File: roles/istio/tasks/certificates.yaml [hunk 2]
```diff
@@ -1,24 +1,8 @@
...
-    (not server_certs_paths_defined and not self_signed_ca) or
-    (server_certs_paths_defined and self_signed_ca)
-
-
-- name: Create temporary certificates directory
-  ansible.builtin.tempfile:
-    state: directory
-    suffix: certificate
-  register: tmp_cert_dir
+    msg: "When not using self signed certificates the certificates paths must be provided."
```
Comment (RafalSkolasinski) [inline roles/istio/tasks/certificates.yaml:4]: In logs we will see title of task so does not need to be so verbose

File: roles/istio/tasks/certificates.yaml [hunk 3]
```diff
@@ -1,24 +1,8 @@
 ---
-- name: Register ssl config options
```
Comment (RafalSkolasinski) [inline roles/istio/tasks/certificates.yaml:2]: no needed

File: roles/istio/tasks/certificates.yaml [hunk 4]
```diff
@@ -36,21 +20,34 @@
...
+  register: tmp_cert_dir
+  when: istio_ssl.self_signed_ca | bool
+
+
 - name: Generate Self Signed Server SSL Certs from Self Signed CA
   include: self_signed.yaml
-  when: self_signed_ca
+  when: istio_ssl.self_signed_ca | bool
 
+- name: Set fact about Server SSL Certs Paths (self-signed certificates)
```
Comment (RafalSkolasinski) [inline roles/istio/tasks/certificates.yaml:40]: will be clearer in logs which one is used this way

File: roles/istio/defaults/main.yaml (hunk 1 above)
Comment (michaelcheah) [reply roles/istio/defaults/main.yaml:8]: I had originally tried out `default(omit)`

File: roles/istio/tasks/certificates.yaml [hunk 5]
```diff
@@ -1,24 +1,8 @@
...
-    (server_certs_paths_defined and self_signed_ca)
-
-
-- name: Create temporary certificates directory
-  ansible.builtin.tempfile:
-    state: directory
-    suffix: certificate
-  register: tmp_cert_dir
+    msg: "When not using self signed certificates the certificates paths must be provided."
+  when: not (istio_ssl.self_signed_ca | bool) and (istio_ssl.server_crt_path is undefined or istio_ssl.server_key_path is undefined or istio_ssl.server_crt_path is none or istio_ssl.server_key_path is none)
```
Comment (michaelcheah) [inline roles/istio/tasks/certificates.yaml:5]: What about the situation where someone sets both `istio_ssl.self_signed_ca: true` AND `istio_ssl.server_crt_path` and `istio_ssl.server_key_path` are set

One needs to take precedence over the other? (I think I originally I checked the current one you've got and this other one)

**Review by michaelcheah — COMMENTED** [review]
Just some questions about how the paths are determined if the configs are set for both options

File: roles/istio/tasks/certificates.yaml (hunk 5 above)
Comment (RafalSkolasinski) [reply roles/istio/tasks/certificates.yaml:5]: I would see `istio_ssl.self_signed_ca` to take precedence. If it is set to `true` then we generate self-signed certs and use them. If set to `false` then we check if paths are provided.

**Comment by RafalSkolasinski** [comment]
Actually with `istio_ssl` being hashmap it was not possible to just provide (assuming changing `self_signed_ca` default to false)
```
istio_ssl:
  enabled: true
  server_crt_path: /tmp/ansible.s7572b4xcertificate/server.crt
  server_key_path: /tmp/ansible.s7572b4xcertificate/server.key
```
because `self_signed_ca` was undefined. 

The way it is now it is possible to either provide 

```
istio_use_ssl: true
istio_use_self_signed_ca: true
```

or 

```
istio_use_ssl: true
istio_ssl_certificates:
  server_crt_path: /tmp/ansible.s7572b4xcertificate/server.crt
  server_key_path: /tmp/ansible.s7572b4xcertificate/server.key
```


with `istio_use_self_signed_ca` just taking priority

File: roles/istio/defaults/main.yaml (hunk 1 above)
Comment (RafalSkolasinski) [reply roles/istio/defaults/main.yaml:10]: unfortunately that doesn't work in this situation

**Comment by RafalSkolasinski** [comment]
Simplified logic. Now vars are as follows:

```
istio_use_ssl: true
istio_self_signed_certs: "{{ istio_use_ssl }}"

istio_seldon_gateway: "{{ lookup('template', 'default-seldon-gateway.yaml.j2') | from_yaml }}"
```


Use case:

1. use just set `istio_use_ssl: true` or `istio_use_ssl: false` -> this by default uses selfed-signed certificates 
2. user set `istio_use_ssl: true` but `istio_self_signed_certs: false` - > same gateway is created but `istio-ingressgateway-certs` must be created by user
3. If the gateway configuration in this two options are not enough one can provide their own definition inline of with custom template `istio_seldon_gateway: "{{ lookup('template', 'my-gateway.yaml.j2') | from_yaml }}"`

**Review by michaelcheah — APPROVED** [review]

**Pull request merged**